nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -472,9 +472,13 @@ void nano::bootstrap_server::bootstrap_message_visitor::bulk_pull (const nano::b
 		server->node->logger.try_log (boost::str (boost::format ("Received bulk pull for %1% down to %2%, maximum of %3% from %4%") % message.start.to_string () % message.end.to_string () % message.count % server->remote_endpoint));
 	}
 
-	// TODO: Add completion callback to bulk pull server
-	auto bulk_pull_server = std::make_shared<nano::bulk_pull_server> (server, std::make_unique<nano::bulk_pull> (message));
-	bulk_pull_server->send_next ();
+	server->node->bootstrap_workers.push_task ([server = server, message = message] () {
```

Comment:
Should server be a weak_pointer which becomes strong inside the callback?

Reply:
Reference to bootstrap_server is only ever stored as weak_ptr in other places so we need to keep it non weak, otherwise it will get destroyed